scrypto/src/engine/api.rs

```diff
@@ -31,7 +31,10 @@ macro_rules! native_functions {
 
 #[derive(Debug, TypeId, Encode, Decode)]
 pub enum RadixEngineInput {
-    Invoke(FnIdent, Vec<u8>),
+    InvokeScryptoFunction(ScryptoFunctionIdent, Vec<u8>),
+    InvokeScryptoMethod(ScryptoMethodIdent, Vec<u8>),
+    InvokeNativeFunction(NativeFunctionIdent, Vec<u8>),
+    InvokeNativeMethod(NativeMethodIdent, Vec<u8>),
```

Comment:
Would rather see this as a single entry. Want to keep abstraction about which interpreter to call outside of this interface.

Reply:
I've left a TODO to combine `fn_ident` and `args` into something like `ScryptoMethodInvocation`. I will do it after this PR.

Reply:
Actually, let me do that within this PR. Doesn't seem to be too big a change.

Reply:
Looks like I misunderstood the question. Are you proposing we have an enum that captures all four variants OR we combine the ident with the args?